### PR TITLE
Add/filter owner events

### PR DESCRIPTION
**Pull request by v1b3m**
ISSUES CLOSED: #65

### CODE REVIEW

**Comment by kevindigo** [comment]
Quick note: Dani (the primary maintainer) and I are attending a conference yesterday and today. We'll review this as soon as we have a chance.

**Comment by v1b3m** [comment]
> Quick note: Dani (the primary maintainer) and I are attending a conference yesterday and today. We'll review this as soon as we have a chance. 

I thought I had been ignored 🤣

File: .env.template
Comment (danisyellis) [inline .env.template:5]: let's change this to IGNORE_SELFOWNED_EVENTS because that's a little bit clearer. (And then, change the variable name everywhere else it's used- the README and in index.js)

And let's set it equal to an empty string "" because this is an option and not the default functionality for Starfish.

File: .prettierrc
Comment (danisyellis) [inline .prettierrc:6]: my boss wants to keep the trailing comma, so let's get rid of this change, please.

File: README.md
Comment (danisyellis) [inline README.md:119]: Thank you for updating the README to reflect the change you made in the code :-)
This sentence should be a bullet point like the others (for example, `- Paste the access token into GITHUB_TOKEN`) to make it easy to find.

Unfortunately, it looks like your file changes almost every line of the README. Maybe Starfish doesn't have linter rules set up for Markdown files, but your text editor does? I actually spent some time looking through the READMEs to see if there was any other content you changed, or if the two files look different, but I think all of these other changes are just how the code in the file itself looks. I don't want to  make everybody else working on Starfish have to update their code so please disable your linter or whatever you need to do so that the only change in the README file is the adding of your instructions.

File: index.js
Comment (danisyellis) [inline index.js:48]: I'd like this function name to mention that it's filtering something *out*. So you could call it something like `filterOutByAuthorAssociation` or `rejectByAuthorAssociation`.

Comment (danisyellis) [inline index.js:48]: I don't think there's a reason that `events` needs a default of `[]`. Events will always be passed in with a value, so I don't think that default would ever get hit and we can take it out.

File: index.js
Comment (danisyellis) [inline index.js:49]: I would expect to see this implemented using `.filter` instead of `.reduce`. Could you switch to use filter or is there a reason you chose reduce instead? Although it works, using reduce adds a lot more code complexity which makes it harder to read and understand.

File: index.js
Comment (danisyellis) [inline index.js:92]: Since we're allowing both true and True, let's also allow TRUE and any other version of the word true that a user might put in. 
An easy way to do that would be to use the javascript function `toLowerCase()` instead of RegEx, but you could also write a more inclusive RegEx.

**Review by danisyellis — CHANGES_REQUESTED** [review]
Thanks for your PR!
I'm excited that we're going to be able to filter out self-owned events.

This past week was difficult to carve out time for triaging all of the Stafish PRs, but I'll be much more responsive this week to make sure everyone can get their contributions in before the end of Hacktoberfest.

I've made a bunch of notes for changes we'd like to see before merging in your PR. Let me know if you have any questions.

**Comment by danisyellis** [comment]
Hi @v1b3m Since we haven't heard back from you about making these changes, we're going to take over the PR, keep what you did, and add some more commits to fix the things mentioned in my code review. 
Thanks again for getting this functionality added to Starfish :-)

**Comment by IngridGdesigns** [comment]
I tested the function FilterByAuthorAssociation() by writing results to a json file and checking author_association (fs.writeFile('./file.json', JSON.stringify(events, null, 2), { encoding: 'utf8' }, (err) => console.log(err)) )
I also extracted some of the repeated code and created a helper function to improve readability.

File: README.md
Comment (kevindigo) [reply README.md]: For what it's worth, I see a ton of linting issues when I load README into vscode. In a separate ticket, we should probably either do a one-pass edit of README to comply with the linting rules, OR change the linting rules to match the current README so there are no warnings OR add the README to the linter's ignore file.

**Review by kevindigo — CHANGES_REQUESTED** [review]
@danisyellis:  I added rocketship emoji responses to the code review comments that I believe have been addressed. I didn't review the README changes. 

@IngridGdesigns: I added a couple new code review comments, including one where I'm pretty sure the code isn't correct (even if it might end up achieving the desired behavior).

File: README.md
Comment (danisyellis) [inline README.md:22]: Currently, your PR can't be merged because of a merge conflict in the README. It's probably because this Hacktoberfest info is gone now. There might be other things that have changed too.
If you look at the bottom of your PR, you'll see that Github tells you it can't be merged and prompts you to resolve the conflicts.

File: README.md
Comment (danisyellis) [inline README.md:119]: now that you've changed the code to accept any capitalization of the word "true", let's make this sentence simpler by only including one version of the word.
"simply set the IGNORE_SELFOWNED_EVENTS environment variable to "true", otherwise

File: README.md
Comment (danisyellis) [reply README.md]: I created a new ticket for that, but then I thought-  do you think that means that we should actually keep all of these README changes because they're fixing those linting issues you're seeing? I'm not seeing them on my computer (though I'm currently in the Andre tests branch.)

Comment (kevindigo) [reply README.md]: I would prefer for this MR to contain a minimum of README changes. Since you already have a README ticket planned, we should probably figure out the linting as part of that.

File: index.js
Comment (kevindigo) [inline index.js:70]: This should either return true, or false. I suspect it should be false, but double check that.

File: index.js
Comment (kevindigo) [inline index.js:49]: It looks like this function is never called with an association parameter. Should we just remove the parameter, and hard-code 'OWNER' in the if statement? It would make the logic much easier to parse.

File: index.js
Comment (kevindigo) [inline index.js:50]: This is a bit of an odd function, in that it returns true if the if was false, and false if the if was true. 

In some cases, to help make it clearer what is going on, I'll extract a local. In this case, that might be (assuming we don't need the association parameter, per my other review comment): 
`const isAuthorAlsoTheOwner = (eventType.author_association === 'OWNER');`

Then the if would be `if (!isAuthorAlsoTheOwner) { return true; }`

Or, at that point, we could eliminate the if and just have:
`return !isAuthorAlsoTheOwner;`

**Pull request merged**
